hub/core/storage/s3.py

```diff
@@ -225,6 +224,63 @@ def __getitem__(self, path):
         except Exception as err:
             raise S3GetError(err) from err
 
+    def _get_bytes(self, path, start_byte: int = None, end_byte: int = None):
+        range = None
+        if start_byte != None and end_byte != None:
+            assert end_byte is not None
+            range = f"bytes={start_byte}-{end_byte - 1}"
+        elif start_byte != None:
+            range = f"bytes={start_byte}-"
+        elif end_byte != None:
+            assert end_byte is not None
+            range = f"bytes=-{end_byte - 1}"
+        resp = self.client.get_object(Bucket=self.bucket, Key=path, Range=range)
+        return resp["Body"].read()
+
+    def get_bytes(
```

Comment:
what about datasets that are stored on local storage? we should probably add get_bytes to local provider as well.

also, what about in memory datasets? maybe we should raise an error if a user tries to visualize those

Reply:
There is generic implementation of the get_bytes, which reads the whole resource(file) and returns part of it. I thought this implementation is enough performant for the local files.

Regarding to the in-memory datasets, I was thinking they have special storage provider which would be runtime dictionary. If that is not the case, then let's discuss to figure our the right way to visualize them too.